viv-wallet-app/src/views/ConsultantEdit.vue

```diff
@@ -0,0 +1,195 @@
+                    <label id="input-endDate-1" label-for="endDate-1">Départ</label>
+                    <input id="endDate-1" type="date" v-model="consultant.endDate" placeholder="Date de départ" />
+                </div>
+
+                <div class="buttons">
+                    <button class="primary-button" v-on:click="confirm">Confirmer</button>
+                    <router-link class="secondary-button" v-bind:to="`/members/${expertiseName}`" tag="button"
+                        >Cancel</router-link
+                    >
+                </div>
+            </form>
+        </loading>
+    </div>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue, Watch } from "vue-property-decorator";
+import { Consultant, ConsultantStatus, toString } from "../models/consultant";
+import { ConsultantsService } from "../services/consultants";
+import { UsersService } from "../services/users";
+import { User } from "../models/user";
+import Loading from "../components/Loading.vue";
+
+const ConsultantEditProps = Vue.extend({
+    props: {
+        expertiseName: String,
+        consultantId: String
+    }
+});
+
+@Component({
+    name: "consultant",
+    components: { Loading }
+})
+export default class ConsultantEdit extends ConsultantEditProps {
+    consultantsService;
+    usersService;
+    consultant: Consultant = {};
+    usersNotAlreadyInExpertise: User[] = [];
+    loading = false;
+    errored = false;
+    options: { text: string; value: string; disabled: boolean }[] = [
+        { text: "Choisissez une option", value: "", disabled: true }
+    ];
+
+    constructor() {
+        super();
+        this.consultantsService = new ConsultantsService(this.expertiseName);
+        this.usersService = new UsersService();
+
+        for (const [key, value] of Object.entries(ConsultantStatus)) {
+            if (isNaN(Number(key))) {
+                // .entries contains either the constant names and indexes
+                this.options.push({
+                    text: toString(value as ConsultantStatus),
+                    value: value as string,
+                    disabled: false
+                });
+            }
+        }
+    }
+
+    async mounted() {
+        try {
+            this.loading = true;
+            if (this.consultantId !== "add") {
+                this.loading = true;
+                this.consultant = await this.consultantsService.getConsultant(this.consultantId);
+            } else {
+                this.loading = true;
+                const consultantsInExpertise = await this.consultantsService.getConsultants();
+                const allUsers = await this.usersService.getUsers();
+
+                this.usersNotAlreadyInExpertise = allUsers.filter(
+                    u => !consultantsInExpertise.find(u1 => u1.id === u.id)
+                );
+                this.consultant.startDate = new Date().toLocaleDateString("en-CA"); // the format should be YYYY-MM-DD
```

Comment:
Not a huge fan of this hack. It's a sign that we should use a real date library (date-fns is light and efficient library). But we can live with it for now.

Reply:
me neither... but I don't know which date format they are going o send us, I need to test it with the real backend.
but I still prefer a one line hack to a full library / dependency :)